lib/full-site-editing/class-gutenberg-rest-templates-controller.php

```diff
@@ -340,18 +340,10 @@ protected function prepare_item_for_database( $request ) {
 		if ( null === $template ) {
 			$changes->post_type   = $this->post_type;
 			$changes->post_status = 'publish';
-			$changes->tax_input   = array(
-				'wp_theme' => isset( $request['theme'] ) ? $request['theme'] : wp_get_theme()->get_stylesheet(),
-			);
 		} elseif ( 'custom' !== $template->source ) {
-			$changes->post_name   = $template->slug;
```

Comment:
This removal caused the unit test failure in `phpunit/class-gutenberg-rest-template-controller-test.php`'s `test_update_item()`. The reason is that upon creating the new `wp_template` post object (with its title [set to `My new Index Title`](https://github.com/WordPress/gutenberg/blob/9aa9d75ec90b303f5dcf93462971ab4f7067eb81/phpunit/class-gutenberg-rest-template-controller-test.php#L244)), we're no longer using the template (file)'s slug (`index`), but creating a slug based on the post title -- in this case, `my-new-index-title`. That slug is then used in the theme mod as the key for the newly created post object's ID -- something like

```php
array(
    'my-new-index-title` => 6
)
```

This means that `gutenberg_get_block_template()` (which is e.g. used for the network response) cannot find a `wp_template` post object for the slug it was given (`index`, via the template ID `tt1-blocks//index`).

I think the right fix is to add this line (and [L354](https://github.com/WordPress/gutenberg/pull/31971/files#diff-c8f55bd6935940e0c1da0ed126bb7642883fc57d1cd5e5b002548c45ce08c789L354)) back -- I don't quite see why they were removed in the first place, so I hope that adding them back won't have any ripple effects. cc/ @carlomanf

Reply:
I think I removed that because I was only thinking about creating new templates, and not the possibility of overwriting a file-based one via PUT request. I can't think of any reason not to add back those lines for the existing endpoint, but they may have been needed for the new endpoint that is not being back-ported to 5.8?

Reply:
What about adding back this line only if `true === $template->has_theme_file`?

Reply:
> What about adding back this line only if `true === $template->has_theme_file`?

I think the `elseif ( 'custom' !== $template->source )` check on R343 is pretty much synonymous to that, no? (Not entirely sure about the `else` branch, where we also had `$changes->post_name = $template->slug;` 🤔 )

Reply:
I think it can be removed from the else branch, because it would no longer be required to stop the front-end from resolving differently as a result of a slug change.

Reply:
Looks like all the tests pass, so I guess it's safe to resolve this thread.